include/envoy/event/scaled_range_timer_manager.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/event/range_timer.h"
+#include "envoy/event/timer.h"
+
+namespace Envoy {
+namespace Event {
+
+/**
+ * Class for creating RangeTimer objects that can be adjusted towards either the minimum or maximum
+ * of their range by the owner of the manager object. Users of this class can call createTimer() to
+ * receive a new RangeTimer object that they can then enable or disable at will (but only on the
+ * same dispatcher), and setScaleFactor() to change the scaling factor. The current scale factor is
+ * applied to all timers, including those that are created later.
```

Comment:
I think the current wording makes it unclear if updates to scale apply to timers that are already created.

Consider: Updates to the current scale factor are applied to all timers, including those created in the past.